lib/zuora/api.rb

```diff
@@ -11,24 +11,47 @@ module Zuora
   # @return [Config]
   def self.configure(opts={})
     Api.instance.config = Config.new(opts)
-    HTTPI.logger = opts[:logger]
-    HTTPI.log = opts[:logger] ? true : false
-    Savon.configure do |savon|
-      savon.logger = opts[:logger]
-      savon.log = opts[:logger] ? true : false
+    if Api.instance.config.sandbox
+      Api.instance.sandbox!
+    elsif Api.instance.config.services
+      Api.instance.set_endpoint Api.instance.config.custom_url
     end
   end
 
   class Api
+    I18n.enforce_available_locales = false
+
     # @return [Savon::Client]
-    attr_accessor :client
+    def client
+      @client ||= make_client
+    end
 
     # @return [Zuora::Session]
     attr_accessor :session
 
     # @return [Zuora::Config]
     attr_accessor :config
 
+    # Zuora::API Config options
+    # @return [Hash]
+    attr_accessor :options
+
+    # Leaving the old approach used to define value of WSDL
+    # till we understand how/where config.wsdl_path is been set
+    # WSDL = if config && config.wsdl_path
+    #          config.wsdl_path
+    #        else
+    #          File.expand_path('../../../wsdl/zuora.a.38.0.wsdl', __FILE__)
+    #        end
+
+    WSDL = File.expand_path('../../../wsdl/zuora.a.78.0.wsdl', __FILE__)
```

Comment:
We should keep the above `config.wsdl_path` code, and update the default wsdl file:
```suggestion
    WSDL = if config && config.wsdl_path
             config.wsdl_path
           else
             File.expand_path('../../../wsdl/zuora.a.78.0.wsdl', __FILE__)
           end
```

Reply:
Resolved here [48578b7](https://github.com/Sage/zuora/pull/18/commits/48578b78b98f6554e58e5840ef2baad3bbd06554)